utils/gear-hooks/src/hooks/sails/use-program-event.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from 'react';
+import { EventReturn, FunctionName, ServiceName, Event, EventCallbackArgs } from './types';
+
+type UseProgramEventParameters<TProgram, TServiceName, TFunctionName, TCallbackArgs extends any[]> = {
+  program: TProgram | undefined;
+  serviceName: TServiceName;
+  functionName: TFunctionName;
+  onData: (...args: TCallbackArgs) => void;
```

Comment:
NIT: You might consider renaming `onData` variable to `onEvent` or `onReceiveEvent`

Reply:
I believe `onData` is more suitable.

Under the hood there's generic `GearEvent` subscription that just awaits data under suitable conditions, therefore `onEvent` might be misleading.